src/nupic/algorithms/Connections.cpp

```diff
@@ -303,10 +306,9 @@ void Connections::updateSynapsePermanence(Synapse synapse,
       potentialPreseg.push_back( segment );
     }
 
-    for (auto h : eventHandlers_) {
+    for (auto h : eventHandlers_) { //TODO handle callbacks in performance-critical method only in Debug?
```

Comment:
this method is a hotspot I found in profiling, do we really need the callbacks here, can be left out only for debug,.. ?

Reply:
Yea I have no idea why these exist but their part of the public API so I didn't remove them.
Maybe try: `for( const auto & : eventHandlers_)` for speed?

Reply:
I'll profile w/o the loop, imo this would be a candidate to break the API, if shows performance gain.